### PR TITLE
Remove face detector from iOS Expo Go

### PR DESCRIPTION
**Pull request by aleqsio**
# Why
https://github.com/expo/fyi/blob/main/face-detector-removed.md

We want to add support for M1 simulator builds and this is a single blocking dependency.
https://issuetracker.google.com/issues/178965151
It's relatively unused and there are documented ways to use it with development builds and prebuild.

# How

Removed it from Expo Go iOS.

**This is a breaking change**
It removes the native dependencies from previous SDKs in versioned code. 

Unfortunately there's no way to do this for simulator only without adding additional logic on top of xCode build system. 
We keep it included in Android Expo Go.

# Test Plan

Tested manually if using the package in Expo Go iOS throws readable errors with more info.

```
 WARN  ExpoFaceDetector has been removed from Expo Go. To use this functionality, you'll have to create a development build or prebuild using npx expo run:android|ios commands.

Learn more: https://expo.fyi/face-detector-removed

Learn more about development builds: https://docs.expo.dev/develop/development-builds/create-a-build/

Learn more about prebuild: https://docs.expo.dev/workflow/prebuild/
```

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-face-detector/src/FaceDetector.ts [hunk 1]
```diff
@@ -204,7 +203,15 @@ export async function detectFacesAsync(
   uri: string,
   options: DetectionOptions = {}
 ): Promise<DetectionResult> {
-  if (!ExpoFaceDetector.detectFaces) {
+  if (!ExpoFaceDetector || !ExpoFaceDetector.detectFaces) {
```
Comment (Kudo) [inline packages/expo-face-detector/src/FaceDetector.ts:206]: would be good if we can further check the running environment is expo go. we have the information from [expo-constants](https://docs.expo.dev/versions/latest/sdk/constants/#executionenvironment). but that would require dependency to expo-constants.
does it make sense to you?

**Review by Kudo — COMMENTED** [review]
we should also add expo-face-detector to unversionable list: https://github.com/expo/expo/blob/main/tools/src/versioning/ios/unversionablePackages.json

thanks for taking care all the changes 👏

File: packages/expo-face-detector/src/FaceDetector.ts (hunk 1 above)
Comment (tsapeta) [reply packages/expo-face-detector/src/FaceDetector.ts:206]: Even though people already have expo-constants installed (expo depends on it), I would rather leave it as is. Adding a dependency always brings the risk of having multiple versions.

**Review by tsapeta — APPROVED** [review]
Looks good to me, besides Kudo's comment about EXFileSystem

File: ios/versioned-react-native/ABI46_0_0/Expo/ExpoKit/ABI46_0_0ExpoKit.podspec [hunk 2]
```diff
@@ -74,8 +74,6 @@ Pod::Spec.new do |s|
     ss.dependency         "ABI46_0_0EXDocumentPicker"
     ss.dependency         "ABI46_0_0EASClient"
     ss.dependency         "ABI46_0_0EXErrorRecovery"
-    ss.dependency         "ABI46_0_0EXFaceDetector"
-    ss.dependency         "ABI46_0_0EXFileSystem"
```
[the remark this answers is not in the document]
Comment (aleqsio) [reply ios/versioned-react-native/ABI46_0_0/Expo/ExpoKit/ABI46_0_0ExpoKit.podspec:78]: no it wasn't, sorry!

File: packages/expo-face-detector/src/FaceDetector.ts (hunk 1 above)
Comment (Kudo) [reply packages/expo-face-detector/src/FaceDetector.ts:206]: checking `global.expo?.modules?.ExponentConstants` is clever 👍

Comment (aleqsio) [reply packages/expo-face-detector/src/FaceDetector.ts:206]: checking appOwnership before showing that warn

**Pull request merged**